adoorback/account/models.py

```diff
@@ -316,11 +317,23 @@ def most_recent_update(self, user):
             return None
 
     def can_access_response_set(self, user):
-        # return responses of user that self can access
         from qna.models import Response
-        response_ids = [response.id for response in user.response_set.all() if Response.is_audience(response, self)]
-        response_queryset = Response.objects.filter(id__in=response_ids)
-        return response_queryset
+        from category.models import Subscription
+
+        #gets the user's subscribed categories
+        subscribed_categories = Subscription.objects.filter(user=self).values_list('category_id', flat=True)
+
+        # Then, filters responses by:
+        # 1. Author (user parameter)
+        # 2. Category (must be in subscribed categories)
+        # 3. Sharing scope (user must be included in sharing scope)
+        responses = Response.objects.filter(
+            author=user,
+            category_id__in=subscribed_categories
```

Comment:
Since _not subscribing to a category_ only means `not getting notifications from the category`,
(not that the user will be unable to access the category)
I think the filtering by `subscribed_categories` is unnecessary here.

Rather, you should filter by whether `self` is in the `sharing_scope` (that the `user` has set) of the category.

Reply:
Got it! I updated my logic to account for this; I think this should filter based on sharing scope rather than subscribed categories now. I think the logic should work now to catch whether it only filters based on whether the current user (self) is included in the sharing scope of each response, which I believe should be the intended implementation!